packages/commonwealth/client/scripts/controllers/chain/ethereum/generalContracts.ts

```diff
@@ -0,0 +1,227 @@
+import $ from 'jquery';
+import m from 'mithril';
+import app from 'state';
+import { Contract, NodeInfo, IWebWallet } from 'models';
+import { initAppState } from 'app';
+import { Contract as Web3Contract } from 'web3-eth-contract';
+import { RLPEncodedTransaction, TransactionConfig, TransactionReceipt } from 'web3-core/types';
+import { parseAbiItemsFromABI, parseEventFromABI } from 'helpers/abi_utils';
+import { AbiItem } from 'web3-utils';
+import { ChainBase, ChainNetwork, ChainType } from 'common-common/src/types';
+import { linkExistingAddressToChainOrCommunity } from 'controllers/app/login';
+import {
+  ChainFormFields,
+  EthFormFields,
+} from 'views/pages/create_community/types';
+import Web3 from 'web3';
+
+type EthDaoFormFields = {
+  network: ChainNetwork.Ethereum;
+  tokenName: string;
+};
+type CreateFactoryEthDaoForm = ChainFormFields &
+  EthFormFields &
+  EthDaoFormFields;
+
+export default class GeneralContractsController {
+  public web3: Web3;
+  public contract: Contract;
+  public web3Contract: Web3Contract;
+  public isFactory: boolean;
+
+  constructor(web3: Web3, contract: Contract) {
+    this.isFactory = contract.isFactory;
+    this.web3 = web3;
+    this.contract = contract;
+    try {
+      this.web3Contract = new this.web3.eth.Contract(
+        parseAbiItemsFromABI(this.contract.abi),
+        this.contract.address
+      );
+    } catch (error) {
+      console.error('Failed to create DaoFactory controller', error);
+    }
+  }
+
+  private async contractCall(web3: Web3, tx: TransactionConfig): Promise<string> {
```

Comment:
Consolidated them. But we should have a retro on all this later because this separation was intentional. It was following the original spec that was created for handing these web3 tx calls.